docs/src/main/asciidoc/security-webauthn.adoc

```diff
@@ -912,22 +863,87 @@ webAuthn.loginOnly({ name: userName })
 == Handling login and registration endpoints yourself
 
 Sometimes, you will want to ask for more data than just a username in order to register a user,
-or you want to deal with login and registration with custom validation, and so the WebAuthn callback
-endpoint is not enough.
+or you want to deal with login and registration with custom validation, and so the default WebAuthn login
+and registration endpoints are not enough.
 
-In this case, you can use the `WebAuthn.loginOnly` and `WebAuthn.registerOnly` methods from the JavaScript
+In this case, you can use the `WebAuthn.loginClientSteps` and `WebAuthn.registerClientSteps` methods from the JavaScript
 library, store the authenticator data in hidden form elements, and send them as part of your form payload
 to the server to your custom login or registration endpoints.
 
-If you are storing them in form input elements, you can then use the `WebAuthnLoginResponse` and
-`WebAuthnRegistrationResponse` classes, mark them as `@BeanParam` and then use the `WebAuthnSecurity.login`
-and `WebAuthnSecurity.register` methods to replace the `/q/webauthn/callback` endpoint. This even
-allows you to create two separate endpoints for handling login and registration at different endpoints.
+If you are storing them in form input elements, you can then use the link:{webauthn-api}/io/quarkus/security/webauthn/WebAuthnLoginResponse.html[`WebAuthnLoginResponse`] and
+link:{webauthn-api}/io/quarkus/security/webauthn/WebAuthnRegistrationResponse.html[`WebAuthnRegistrationResponse`] classes, 
+mark them as `@BeanParam` and then use the 
+link:{webauthn-api}/io/quarkus/security/webauthn/WebAuthnSecurity.html#login(io.quarkus.security.webauthn.WebAuthnLoginResponse,io.vertx.ext.web.RoutingContext)[`WebAuthnSecurity.login`]
+and link:{webauthn-api}/io/quarkus/security/webauthn/WebAuthnSecurity.html#register(io.quarkus.security.webauthn.WebAuthnRegisterResponse,io.vertx.ext.web.RoutingContext)[`WebAuthnSecurity.register`]
+methods to replace the `/q/webauthn/login` and `/q/webauthn/register` endpoints.
 
-In most cases you can keep using the `/q/webauthn/login` and `/q/webauthn/register` challenge-initiating
+In most cases you can keep using the `/q/webauthn/login-challenge` and `/q/webauthn/register-challenge` challenge-initiating
 endpoints, because this is not where custom logic is required.
 
-For example, here's how you can handle a custom login and register action:
+In this case, the registration flow is a little different because you will write your own registration endpoint
+which will handle storing of the credentials and setting up the session cookie:
+
+image::webauthn-custom-register.svg[role="thumb"]
+
+Similarly, the login flow is a little different because you will write your own login endpoint
```

Comment:
@FroMage  What is the rationale for letting users control the login process ?

Reply:
As stated a few lines up:

> Sometimes, you will want to ask for more data than just a username in order to register a user,
> or you want to deal with login and registration with custom validation, and so the default WebAuthn login
> and registration endpoints are not enough.

Is this not compelling enough?